.github/workflows/sdk-python-gate-check.yml

```diff
@@ -0,0 +1,44 @@
+name: sdk-python-gate-check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+permissions:
+  contents: read
+  checks: read
+
+jobs:
+  python_sdk_gate:
+    name: sdk-python-gate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Check for sdk-python path changes
+        id: filter
+        uses: dorny/paths-filter@v3
+        with:
+          filters: |
+            python_sdk:
+              - 'python-sdk/**'
+
+      - name: Log path changes decision
+        run: 'echo "Python SDK files changed: ${{ steps.filter.outputs.python_sdk }}"'
+
+      - name: Wait for python-sdk-ci if changes detected
+        if: steps.filter.outputs.python_sdk == 'true'
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          check-name: 'test' # The job ID from python-sdk-ci.yml
```

Comment:
Maybe we can use a variable here so that it's not a magic string that can be changed by accident?

Reply:
Yeah. I think we should make this a template workflow and then we can easily use it across typescript, python, go, etc